network/p2p/compressed/compressedStream.go

```diff
@@ -0,0 +1,99 @@
+package compressed
+
+import (
+	"fmt"
+	"io"
+	"sync"
+
+	"github.com/libp2p/go-libp2p-core/network"
+	"go.uber.org/multierr"
+
+	flownet "github.com/onflow/flow-go/network"
+	"github.com/onflow/flow-go/network/compressor"
+)
+
+// compressedStream is an internal networking layer data structure,
+// which implements a compression mechanism as a wrapper around a native
+// libp2p stream.
+type compressedStream struct {
+	network.Stream
+
+	writeLock  sync.Mutex
+	readLock   sync.Mutex
+	compressor flownet.Compressor
+
+	r io.ReadCloser
+	w flownet.WriteCloseFlusher
+}
+
+type StreamOptFunc func(*compressedStream)
+
+// WithStreamCompressor is an option function setting the underlying compressor of the stream.
+func WithStreamCompressor(comp flownet.Compressor) StreamOptFunc {
+	return func(stream *compressedStream) {
+		stream.compressor = comp
+	}
+}
```

Comment:
I would argue this should be a required parameter, rather than an optional and having gzip be the default compressor.

It is not clear from the name that `NewCompressedStream` returns a gzip compressor, and I think making the compressor a required parameter in the API makes it more clear.